.github/workflows/async-update.yml

```diff
@@ -0,0 +1,82 @@
+---
+name: Update when async branch is pushed
+
+'on':
+  workflow_call:
+
+permissions:
+  pull-requests: write   # to create labels, see https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#create-a-label
+
+jobs:
+  update:
+    name: Update everything
+    runs-on: ubuntu-latest
+    # only run if the created branch or tag starts with the correct pattern, namely "async-"
+    if: startsWith(github.ref, 'refs/heads/async-v')
+    steps:
+      - name: Update, remove labels, publish table
+        env:
+          GH_TOKEN: ${{ github.token }}
+          ref: ${{ github.ref }}
+
+        id: create_label
+        run: |
+          branch_name="${ref##*/}"
+
+          # we check now if this branch contains the tag <branch-name>-start which is the one where we branched off the default back back in the days
+          start_tag="${branch_name}-start"
+          if ! GIT_DIR=${GITHUB_REPOSITORY} git branch --contains "${start_tag}" | grep "${branch_name}" ; then
+            echo "ERROR: Expected start tag ${start_tag} is not contained in this branch ${branch_name}."
+            exit 1
+          fi
+
+          # remember the commit of the tag from where this branch started originally; use it as break condition
+          commit_of_start_tag=$(git rev-list -n 1 "${start_tag}")
+
+          commits_async_branch="commits.log"
+          # get log with all commits, use --decorate=short to get the tags as well
+          GIT_DIR=${GITHUB_REPOSITORY} git log --decorate=short "${ref}" > "${commits_async_branch}"
+
+          # now we go through the commits
+
+          # remember the current commit on this branch
+          current_commit_sha=
+          # the commit hash it was cherry-picked from
+          linked_commit_sha=
+
+          while read -r line ; do
+            if [[ "${line}" == "commit *" ]] ; then
+
+              # break if we reached the tag where this branch started
+              current_commit_sha_tmp=$(echo "${line}" | cut -f 2 -d ' ')
+              [[ "${current_commit_sha_tmp}" == "${commit_of_start_tag}" ]] && break
+
+              if [[ -n  "${current_commit_sha}" && -z "${linked_commit_sha}" ]] ; then
+                # this means we ran into a new commit without finding a linked SHA in between, break here
+                echo "Publish only this commit for which we don't find a linked commit"
+                # do something
```

Comment:
are you missing something here?

Reply:
Yes, but the question how/where to publish would be something that we would need to discuss.
This is why I only put comments.